.github/workflows/main.yaml

```diff
@@ -0,0 +1,83 @@
+name: CI
+
+on:
+  push:
+    branches: ["main"]
```

Comment:
Don't we want to run the CI on every push to any branch? If someone, for some reason, doesn't run the tests locally, he may be really surprised upon PR creation ;)

Reply:
Every time I push a change to this PR, the CI is running, I think because of `pull_request` trigger.
shouldn't that be enough?
or do you mean running the CI, before creating the PR?

Reply:
Yes, I mean before.

Reply:
I was following the same pattern as some other charms we maintain.
But I will change it :)

Reply:
Not sure which ones, but all the `charmed-magma-orchestator` charms run CI on every push to the branch. An example:
```
name: orc8r-accessd

on:
  push:
    paths:
      - "orc8r-accessd-operator/**"
```

Reply:
What about the one in `main.yaml` in `charmed-magma-orchestrator`?
```python
name: main-branch

on:
  pull_request:
    branches:
      - main
  push:
    branches:
      - main
```
My point here is that I don't always want to run the CI on push, I push to the branch with no PR as "cloud storage" to save changes when it's WIP. And when I have something like a "v0" I open a draft PR to start testing and trigger the CI. Don't know if that makes sense.

Reply:
I'm not saying we have to do it. I'm just asking if this was intentional or not.

In any case, in my opinion, ideally, development should be done in the branch, including the CI, and once developer thinks it's ready, PR should be created. Draft PRs I'd save for special occasions (like making changes in the PR specific workflows). But that's just my opinion. While in our case it's not a big problem to do development in PRs, in case of big and active projects, that would generate dozens of PRs and a huge mess.

Reply:
I agree with you @Gmerold, the developer should be able to run CI before creating the PR :)
Done.